static/js/topic_page/page_selector.tsx

```diff
@@ -0,0 +1,153 @@
+
+interface PageSelectorPropType {
+  selectedPlace: NamedTypedPlace;
+  selectedTopic: string;
+  topicsSummary: TopicsSummary;
+}
+
+export function PageSelector(props: PageSelectorPropType): JSX.Element {
+  const [placeOptions, setPlaceOptions] = useState<
+    Record<string, string> | undefined
+  >({});
+
+  useEffect(() => {
+    getPlaceOptions(props.selectedTopic, props.topicsSummary, setPlaceOptions);
+  }, [props]);
+
+  const topicName =
+    props.topicsSummary.topicNameMap[props.selectedTopic] ||
+    props.selectedTopic;
+
+  return (
+    <div className="page-selector-container">
+      <div className="page-selector-section">
+        <h1>{topicName} in</h1>
+        <div>
+          <CustomInput
+            id="place-selector"
+            type="select"
+            value={props.selectedPlace ? props.selectedPlace.dcid : ""}
+            onChange={(e) => selectPlace(props.selectedTopic, e)}
+            className="pac-target-input"
+          >
+            {_.isEmpty(props.selectedPlace.dcid) ? (
+              <option value="" key="empty">
+                Select a place
+              </option>
+            ) : (
+              <option
+                value={props.selectedPlace.dcid}
+                key={props.selectedPlace.dcid}
+              >
+                {props.selectedPlace.name}
+              </option>
+            )}
+            {placeOptions &&
+              Object.keys(placeOptions)
+                .filter((place) => place !== props.selectedPlace.dcid)
+                .map((place) => (
+                  <option value={place} key={place}>
+                    {placeOptions[place]}
+                  </option>
+                ))}
+          </CustomInput>
+        </div>
+      </div>
+      <div className="page-selector-section">
+        <CustomInput
+          id="topic-selector"
+          type="select"
+          value={props.selectedTopic}
+          onChange={(e) =>
+            selectTopic(props.selectedPlace.dcid, props.topicsSummary, e)
+          }
+          className="pac-target-input"
+        >
+          <option value={props.selectedTopic} key={props.selectedTopic}>
+            {topicName}
+          </option>
+          {Object.keys(props.topicsSummary.topicNameMap)
+            .filter((topic) => topic !== props.selectedTopic)
+            .map((topic) => (
+              <option value={topic} key={topic}>
+                {props.topicsSummary.topicNameMap[topic] || topic}
+              </option>
+            ))}
+        </CustomInput>
+      </div>
+    </div>
+  );
+}
+
+function getPlaceOptions(
+  selectedTopic: string,
+  topicsSummary: TopicsSummary,
+  setPlaceOptions: (placeOptions: Record<string, string>) => void
+): void {
+  const placeOptionDcids = topicsSummary.topicPlaceMap[selectedTopic] || [];
+  const placeParams = placeOptionDcids
+    .map((place) => "dcid=" + place)
+    .join("&");
+  axios
+    .get(`/api/place/name?${placeParams}`)
```

Comment:
we can call this server-side in flask and pass it down (since it's a fixed map) instead of calling it from the client. add a TODO for now?

Reply:
done